### DISCUSSION TITLE
Migration d'une nouvelle version d'un formulaire

### DISCUSSION THREAD

**Discussion by ChrisThe01** [MTESSDev/FRW#259] [Annonces]
Bonjour,

Présentement nous avons des formulaire qui sont en cours de saisie par la clientèle et n'ont pas été soumis.

Nous voulons graduer une nouvelle version du formulaire.

J'aimerais savoir que seront les impacts de graduer une nouvelle version d'un formulaire lorsqu'il est en a qui sont en cours de saisie.

Merci,

**Comment by loima02** [comment]
Bonjour,

Supposons que l'utilisateur complète son formulaire débuté avec l'ancienne configuration et tente de le soumettre, les validations seront faites en tenant compte de la nouvelle version de la configuration.

Si un nouveau champ obligatoire n'est pas saisi, les validations empêcheront la soumission tant que les messages d'erreurs ne seront pas corrigés. L'utilisateur verra alors la nouvelle configuration et les erreurs de validations apparaître de son côté.

Si un champ est supprimé dans la nouvelle configuration, l'information ne sera tout simplement pas conservée du côté serveur.

Dans les deux cas, il n'y aura pas de "breaking change" et la transition se fera en douceur.

Si vous jugez que les modifications entre l'ancien et le nouveau formulaire sont majeures, il est possible de créer un nouveau formulaire pour qu'un ancien formulaire soit traité avec l'ancienne config et un nouveau formulaire passe par la nouvelle config.

Merci et bonne journée!

**Comment by ChrisThe01** [comment]
Merci,

Par contre, dans mon cas c'est un peu plus compliqué. À la suite d'une demande de changement, je dois renommer 3 champs (car notre traitement doit utiliser une nomenclature spécifique pour récupérer ces valeurs).
De plus, il y 4 autres champs qui sont saisie dans un tableau. Ces champs ne sont pas accessible par notre traitement (encore cette fois, une nomenclature spécifique doit être utilisé). J'aimerais pouvoir assigner ces éléments du tableau dans des champs respectant la nomenclature juste avant de soumettre le formulaire.

Si on maintient la situation actuelle, il y aura une perte d'information de saisie pour notre clientèle. Actuellement, il y plus que 70 formulaires qui sont en cors de saisie.

Si possible j'aimerais pourvoir faire une rencontre pour mieux préciser mon problème.

**Reply by ChrisThe01** [reply]
Bonjour,

Finalement, nous avons décidé de lasser les formulaires tel quel même si nous perdons des données.

Alors, on peut fermer cette discussion.

Merci,

**Reply by Hutson48** [reply]
Bonjour,
Pas de problème, revenez-nous si vous avez besoin de quelque chose d'autre!
Bonne journée,
